context.go

```diff
@@ -347,21 +348,42 @@ func (c Context) Interface(key string, i interface{}) Context {
 	return c
 }
 
-type callerHook struct{}
+type callerHook struct{
+	callerSkipFrameCount int
+}
+
+func newCallerHook(skipFrameCount int) callerHook {
+	return callerHook{ callerSkipFrameCount: skipFrameCount}
+}
 
 func (ch callerHook) Run(e *Event, level Level, msg string) {
-	// Three extra frames to skip (added by hook infra).
-	e.caller(CallerSkipFrameCount + 3)
+	switch ch.callerSkipFrameCount {
+	case math.MinInt32:
```

Comment:
For readability, it would be better to set a private const with an opaque value instead of `math.MinInt32`. You can use the same value, just name it.